packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -52,17 +54,30 @@ export class ApexActionController {
             throw new Error(nls.localize('cannot_gather_context'));
           }
 
-          // Step 3: Generate OpenAPI Document
-          progress.report({ message: nls.localize('generate_openapi_document') });
-          const openApiDocument = await this.generateOpenAPIDocument(eligibilityResult, context);
-
-          // Step 4: Write OpenAPI Document to File
+          // Step 3: Determine filename
           name = isClass
             ? path.basename(eligibilityResult.resourceUri, '.cls')
             : eligibilityResult?.symbols?.[0]?.docSymbol?.name;
-          const openApiFileName = `${name}_openapi.yml`;
+          const openApiFileName = `${name}.externalServiceRegistration-meta.xml`;
+          let fullPath;
+          // Step 4: Check if the file already exists
+          try {
+            fullPath = await this.pathExists(openApiFileName);
```

Comment:
@CristiCanizales if we are going to move forward throwing an error if the file already exists, please log a new work item to handle overwrite/cancel.